ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,35 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  try
+    match List.assoc "secureboot" platform with
+    | "auto" -> ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) :: (List.remove_assoc "secureboot" platform)
+    | _ -> platform
+  with Not_found -> platform
+
+let save_uefi_certificates_to_dir ~__context ~pool ~vm =
+  let err () =
+    debug "Error writing uefi certificates to file: varstored directory does not exist or is not a directory";
+    failwith "Failed to save UEFI certificates: varstored directory does not exist or is not a directory"
+  in
+  let uefi_key = !Xapi_globs.varstore_dir ^ "KEK.auth" in
+  if not (Sys.file_exists uefi_key) then
+    if Sys.file_exists !Xapi_globs.varstore_dir && Sys.is_directory !Xapi_globs.varstore_dir then begin
+      let contents = (Xapi_stdext_base64.Base64.decode (Db.Pool.get_uefi_certificates ~__context ~self:pool)) in
+      if contents <> "" then
+        let filename = "xapi_uefi_certificates.tar" in
```

Comment:
Please delete the file after you're finished extracting it.

Reply:
Also would be useful to print a debug message here, so we know we've extracted the certificates.